packaging changes

dodgejoel opened the pull request:
Rebase of 'packaging' branch onto master to get build to trigger and make code mergable. See https://github.com/hlin117/mdlp-discretization/pull/22.

Commit:
packaging changes

Commit:
add in cpp file

hlin117 commented on README.md:
````diff
@@ -9,45 +9,44 @@ for more information.
 
 ```
-pip install -r requirements.txt
+pip install git+https://github.com/$repo_location
````
Change `$repo_location -> hlin117/mdlp-discretization`.

hlin117 commented on README.md:
````diff
@@ -9,45 +9,44 @@ for more information.
 
 ```
-pip install -r requirements-test.txt
+git clone https://github.com/$repo_location
````
This needs to be changed too.

hlin117 commented on README.md:
```diff
@@ -9,45 +9,44 @@ for more information.
+# Development
+
+To submit changes to this project, make sure that you have Cython installed and
+submit the compiled *.cpp file along with changes to python code after running
```
Nice tip!

hlin117 commented on setup.py:
```diff
@@ -1,8 +1,62 @@
+        sources = ['mdlp/_mdlp.cpp']
+
+    class CustomBuildExt(build_ext):
+        """
```
Use PEP 257 here please.
https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

dodgejoel replied:
I rewrote this docstring specifically in response to your previous request and link to PEP 257 and I believe this docstring follows that standard. Can you be more specific about what you would like to be changed?

hlin117 replied:
I meant something like this, sorry if it wasn't clear:

```
"""Custom build_ext class to defer numpy imports until needed.

Overrides the run command for building an extension and adds in numpy
include dirs to the extension build. Doing this at extension build time
allows us to avoid requiring that numpy be pre-installed before
executing this setup script.
"""
```

dodgejoel replied:
sure.

dodgejoel replied:
To be clear. I understand that you are just asking that the docstring start on the same line as the opening quotes, right? Of course I should follow the style of the package and do this but the PEP does not insist on this FYI:

"The summary line may be on the same line as the opening quotes or on the next line."

hlin117 replied:
Hmm, interesting.

hlin117 commented on setup.py:
```diff
@@ -1,8 +1,62 @@
+        from Cython.Distutils import build_ext
+        sources = ['mdlp/_mdlp.pyx']
+    except:
+        # else we build from the cpp file included in the distribution
```
Nice. I appreciate the comments.

hlin117 commented on setup.py:
```diff
@@ -1,8 +1,62 @@
-    include_dirs=[numpy.get_include()]
-)
+"""
+Implements the MDLP discretization criterion from Usama Fayyad's paper
```
You might be interested in this: http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html#example-numpy

hlin117 replied:
I would also recommend making `setup.py` an executable file.

dodgejoel replied:
Are you suggesting that the docstring here is insufficient or just providing that reference for my general enlightenment?

You just want me to add a #!/usr/bin/python at the head of the file?

hlin117 replied:
`chmod +x setup.py` :)

hlin117 commented on setup.py:
```diff
@@ -1,8 +1,62 @@
+
+    setup(
+        name='mdlp-discretization',
+        version='0.1',
```
I'm not officially keeping track of version numbers here... But if I were, I guess we would be at `0.3`?

dodgejoel replied:
Sure.

hlin117 reviewed (changes requested):
This is very close from being merged. I just have the nits mentioned below. Thank you again for your time and effort on this!

Commit:
general cleanup

1. make install link in readme explicit
2. link to paper links to higher quality scan
3. setup.py is executable
4. set version number correctly

hlin117 commented:
Thank you @dodgejoel !

dodgejoel commented:
Yes thank you!
